amazon-kinesis-client/src/test/java/software/amazon/kinesis/config/KCLAppConfig.java

```diff
@@ -0,0 +1,197 @@
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClientBuilder;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClientBuilder;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClientBuilder;
+import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.kinesis.common.ConfigsBuilder;
+import software.amazon.kinesis.common.InitialPositionInStream;
+import software.amazon.kinesis.processor.ShardRecordProcessorFactory;
+import software.amazon.kinesis.retrieval.RetrievalConfig;
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.Optional;
+
+public interface KCLAppConfig {
+
+    String getStreamName();
+
+    default String getStreamArn() {
+        return null;
+    }
+
+    int getShardCount();
+
+    String getApplicationName();
+
+    String getEndpoint();
+
+    Region getRegion();
+
+
+    /**
+     * "default" profile, should match with profiles listed in "cat ~/.aws/config"
+     */
+    String getProfile();
+
+    InitialPositionInStream getInitialPosition();
+
+    Protocol getConsumerProtocol();
+
+    Protocol getProducerProtocol();
+
+    ProducerConfig getProducerConfig();
+
+    ReshardConfig getReshardConfig();
+
+    default KinesisAsyncClient buildConsumerClient() throws URISyntaxException, IOException {
+        return buildAsyncKinesisClient(getConsumerProtocol());
+    }
+
+    default KinesisAsyncClient buildProducerClient() throws URISyntaxException, IOException {
+        return buildAsyncKinesisClient(getProducerProtocol());
+    }
+
+    default KinesisAsyncClient buildAsyncKinesisClient(Protocol protocol) throws URISyntaxException, IOException {
+        return buildAsyncKinesisClient(Optional.ofNullable(protocol));
+    }
+
+    default KinesisAsyncClient buildAsyncKinesisClient(Optional<Protocol> protocol) throws URISyntaxException, IOException {
+
+        /**
+         * Setup H2 client config.
+         */
+        final NettyNioAsyncHttpClient.Builder builder = NettyNioAsyncHttpClient.builder()
+                .maxConcurrency(Integer.MAX_VALUE);
+
+        /**
+         * If not present, defaults to HTTP1_1
+         */
+        if (protocol.isPresent()) {
+            builder.protocol(protocol.get());
+        }
+
+        final SdkAsyncHttpClient sdkAsyncHttpClient =
+                builder.buildWithDefaults(AttributeMap.builder().put(SdkHttpConfigurationOption.TRUST_ALL_CERTIFICATES, true).build());
```

Comment:
Can we remove this [`TRUST_ALL_CERTIFICATES`](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/SdkHttpConfigurationOption.html#TRUST_ALL_CERTIFICATES) so SSL validation is re-enabled? Do the tests execute/pass without this?

Reply:
Yes they do, removing this